src/hackingBuddyGPT/utils/openai/openai_lib.py

```diff
@@ -31,32 +40,116 @@ def client(self) -> openai.OpenAI:
     def instructor(self) -> instructor.Instructor:
         return instructor.from_openai(self.client)
 
-    def get_response(self, prompt, *, capabilities=None, **kwargs) -> LLMResult:
+    def get_response(self, prompt, *, capabilities: Dict[str, Capability]=None, **kwargs) -> LLMResult:
```

Comment:
type information for prompt missing

Reply:
the parent class also has no type information for prompt, what should we limit it to? keep in mind the currently commented out code below that is a compatibility layer to allow both string, renderable and full message list support

Reply:
what would you suggest? I think the code was written by you (:

Reply:
well there was a reason for not adding a type, so that we can vary in the implementations :)